packages/odyssey-react-mui/src/labs/DataComponents/DataView.tsx

```diff
@@ -0,0 +1,387 @@
+  onReorderRows,
+  paginationType = "paged",
+  resultsPerPage = 20,
+  searchDelayTime,
+  stackOptions,
+  tableOptions,
+  totalRows,
+}: DataViewProps) => {
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+
+  const [currentLayout, setCurrentLayout] = useState<Layout>(
+    initialLayout ?? availableLayouts[0],
+  );
+
+  const [data, setData] = useState<MRT_RowData[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(isLoadingProp ?? true);
+  const [isEmpty, setIsEmpty] = useState<boolean>(isEmptyProp ?? true);
+  const [isNoResults, setIsNoResults] = useState<boolean>(
+    isNoResultsProp ?? false,
+  );
+  const [errorMessage, setErrorMessage] =
+    useState<UniversalProps["errorMessage"]>(errorMessageProp);
+
+  const [search, setSearch] = useState<string>("");
+  const [initialFilters, setInitialFilters] = useState<DataFilter[]>();
+  const [filters, setFilters] = useState<DataFilter[]>();
+
+  const [draggingRow, setDraggingRow] = useState<MRT_Row<MRT_RowData> | null>();
+
+  const [rowSelection, setRowSelection] = useState<MRT_RowSelectionState>({});
+
+  useEffect(() => {
+    onChangeRowSelection?.(rowSelection);
+  }, [rowSelection, onChangeRowSelection]);
+
+  const [pagination, setPagination] = useState({
+    pageIndex: currentPage,
+    pageSize: resultsPerPage,
+  });
+
+  const [tableState, setTableState] = useState<TableState>({
+    columnSorting: [],
+    columnVisibility: {},
+    rowDensity: tableOptions?.initialDensity ?? densityValues[0],
+  });
+
+  const shouldShowFilters = useMemo(
+    () => hasSearch || hasFilters,
+    [hasSearch, hasFilters],
+  );
+
+  const availableFilters = useFilterConversion({
+    filters: filters,
+    columns: tableOptions?.columns,
+  });
+
+  const dataQueryParams = useMemo(
+    () => ({
+      page: pagination.pageIndex,
+      resultsPerPage: pagination.pageSize,
+      search,
+      filters,
+      sort: tableState?.columnSorting,
+    }),
+    [pagination, search, filters, tableState?.columnSorting],
+  );
+
+  const getRowId = getRowIdProp ? getRowIdProp : (row: MRT_RowData) => row.id;
+
+  // Set initial filters
+  useEffect(() => {
+    if (!initialFilters && filters) {
+      setInitialFilters(filters);
+    }
+  }, [filters, initialFilters]);
+
+  // Update pagination state if props change
+  useEffect(() => {
+    setPagination({
+      pageIndex: currentPage,
+      pageSize: resultsPerPage,
+    });
+  }, [currentPage, resultsPerPage]);
+
+  // Retrieve the data
+  useEffect(() => {
+    fetchData({
+      getDataFn,
+      setIsLoading,
+      setErrorMessage,
+      errorMessageProp,
+      setData,
+      dataQueryParams,
+    });
+  }, [dataQueryParams, errorMessageProp, getDataFn]);
+
+  // When data is updated
+  useEffect(() => {
+    setIsEmpty(
+      pagination.pageIndex === currentPage &&
+        pagination.pageSize === resultsPerPage &&
+        search === "" &&
+        filters === initialFilters &&
+        data.length === 0,
+    );
```

Comment:
Isn't it possible to be empty even if you have a search term in there? Or am I missing something?

Reply:
We keep track of two states like this — `isEmpty` and `isNoResults`.
- `isNoResults` is for when the data is filtered or searched and the returned value is empty.
- `isEmpty` is for when the data is _**not**_ filtered or searched and the returned value is empty. In other words, it's for when there's no data to be found at all; it's an empty initial state.

In this case, `setIsEmpty` determines if each of the `fetchData` params is empty.